packages/apm/src/hubextensions.ts

```diff
@@ -57,14 +42,17 @@ function startSpan(context: SpanContext | TransactionContext): Transaction | Spa
     logger.warn('Will fall back to <unlabeled transaction>, use `transaction.setName()` to change it.');
   }
 
-  if ((newSpanContext as TransactionContext).name) {
+  if ((context as TransactionContext).name) {
```

Comment:
Do we mention somewhere that the _only_ thing that makes this method work differently is the existence of `.name` property?

Reply:
Forgot to add a line there ^^
`Sentry.startTransaction` has the interface of receiving a `TransactionContext` which has the name property required. This is a safeguard for JS usage.